index.html

```diff
@@ -1,100 +1,122 @@
-          var worksheet = workbook.Sheets[workbook.SheetNames[0]];
-          var selectedCell = worksheet[XLSX.utils.encode_cell({ r: selectedOption, c: 5 })];
-          var cellValue = selectedCell ? selectedCell.v : 'Brak danych';
-          document.getElementById('wybranyokreg').textContent = "Wybrany okreg wyborczy zawiera " + cellValue + " wyborcow";
-        };
-
-        reader.readAsArrayBuffer(loadedFile);
-      } else {
-        alert('Proszę najpierw załadować plik Excel.');
-      }
-    });
-  </script>
+    <h1>liczba wyborcow w danych okregach wyborczych</h1>
+    <button id="przyciskcsv">Załaduj plik CSV</button> <br>
+    <input type="file" id="filei" accept=".csv" style="display: none;">
+    <select id="select"></select>
+    <div id="wybranyokreg"></div>
+    <div id="kwadratodleglosci"></div>
+
+    <script>
+        var loadedFile;
+        var addedDistricts = [];
+
+        document.getElementById('przyciskcsv').addEventListener('click', function () {
+            document.getElementById('filei').click();
+        });
+
+        document.getElementById('filei').addEventListener('change', function () {
+            var file = this.files[0];
+            if (file) {
+                loadedFile = file;
+                alert('Plik CSV został załadowany i zapisany.');
+                populateDropdown();
+            }
+        });
+
+        document.getElementById('select').addEventListener('change', function () {
+            var selectedOption = this.value;
+            if (loadedFile) {
+                var reader = new FileReader();
+                reader.onload = function (e) {
+                    var csvData = e.target.result;
+                    var rows = csvData.split('\n');
+                    var selectedDistrict = rows[selectedOption].split(';')[0];
+
+                    var districtRows = rows.filter(function (row) {
+                    return row.startsWith(selectedDistrict + ';');
+                    });
 
+                    displayResults(districtRows);
+                    kwadratodleglosci();
+                };
 
+                reader.readAsText(loadedFile);
+            } else {
+                alert('Proszę najpierw załadować plik CSV.');
+            }
+        });
 
+        function kwadratodleglosci() {
+            var selectedOption = document.getElementById('select').value;
+            if (loadedFile && selectedOption !== "") {
+                var reader = new FileReader();
+                reader.onload = function (e) {
+                    var csvData = e.target.result;
+                    var rows = csvData.split('\n');
+                    var selectedDistrict = rows[selectedOption].split(';')[0];
+
+                    var districtRows = rows.filter(function (row) {
+                    return row.startsWith(selectedDistrict + ';');
+                    });
+
+                    var squaredDistance = 0;
+                    for (var i = 0; i < districtRows.length; i++) {

```

Comment:
Minor:

Nowadays when iterating through an array, I see people usually [use `for ... of` loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) or [the `array.forEach` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach).

Both of these ways have a major advantage over this regular `for` loop with an `i` index: they are easier conceptually to understand, and there is no need to check for edge cases or mistakes such as `i--` or `i <= districtRows.length`. I suggest using one of these methods:

```ts
for (const row of districtRows) {
  const rowData = row.split(':');
  // the rest of the code
}

// or

districtRows.forEach((row) => {
  const rowData = row.split(':');
  // the rest of the code
});
```

In the second example (the one with `forEach`) I am using [an arrow function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions), which is a shorter way to define a `function` expression.